Makes phpcollection/phpcollection redundant

jurgenhaas opened the pull request:
Fixes #188

jurgenhaas committed:
Fixes #188

jurgenhaas committed:
Fixes #188

GenieTim commented:
Thank you very much for this contribution!
If you want, you can use `composer fix-cs` to automatically fix the code style issues

jurgenhaas committed:
Fixes #188

jurgenhaas committed:
Fixes #188

jurgenhaas committed:
Fixes #188

jurgenhaas committed:
Fixes #188

jurgenhaas commented:
Hmm, that was kind of a challenge. I cloned the repository, ran `composer update` and then called `composer fix-cs` which failed with error messages like this:

```
Call to undefined function grapheme_strlen() in /drupal/vendor/symplify/easy-coding-standard/vendor/symfony/string/UnicodeString.php:171
```

Not sure, why that happened but I replaced `grapheme_strlen` with `mb_strlen` and then the tests worked.

I then realized that the code quality of those classes was really terribly. I went through a couple of iterations and then just had some errors left from classes, that had been there before:

```
Property GitElephant\Objects\Diff\DiffChunk::$originStartLine (int) does not accept string.
```

I fixed them as well, hoping this is not breaking anything. But now it seems that the testbot is running into timeouts. Can you probably force a re-test?
